File: backend/app/schemas/group.py
from pydantic import BaseModel, ConfigDict


class GroupSummary(BaseModel):
    id: int
    code: str
    name: str

    model_config = ConfigDict(from_attributes=True)


class GroupTableRow(BaseModel):
    rank: int
    team_id: int
    team_code: str
    team_name: str
    played: int
    wins: int
    draws: int
    losses: int
    goals_for: int
    goals_against: int
    goal_difference: int
    points: int


class GroupTableResponse(BaseModel):
    group_code: str
    rows: list[GroupTableRow]

File: frontend/src/types/group.ts
export type GroupSummary = {
  id: number;
  code: string;
  name: string;
};

export type GroupTableRow = {
  rank: number;
  team_id: number;
  team_code: string;
  team_name: string;
  played: number;
  wins: number;
  draws: number;
  losses: number;
  goals_for: number;
  goals_against: number;
  goal_difference: number;
  points: number;
};

export type GroupTable = {
  group_code: string;
  rows: GroupTableRow[];
};

